### DISCUSSION TITLE
Population fails 500 error Nestjs

### DISCUSSION THREAD

**Discussion by j-vitali** [Automattic/mongoose#14609] [Q&A]
Hi guys, I'm using nestjs with mongoose and noticed that if a there's a population with an `_id` **which was deleted**, app crashes and says 500 error.
anyone can help me pls?
Thanks.

---
<img width="720" alt="Screenshot 2024-05-23 alle 17 41 20" src="https://github.com/Automattic/mongoose/assets/161858414/4e5c0dc7-b1f2-4e7a-a83b-5eebad6d3002">

My code:

---

 ```
async retrieve(id: string): Promise<any> {
    const document = await this.model
      .findOne({
        _id: id,
      })
      .populate({
        path: 'industry',
        model: this.industryModel,
      })
      .orFail()
      .then((doc: any) => {
        const child = doc.child;
        if (child == null || child instanceof Types.ObjectId) {
          throw new Error('should be populated');
        } else {
          // Works
          doc.child.name.trim();
        }
      });

    if (document === null) {
      return undefined;
    }

    console.log('document', document);

    const org = Organization.fromDocument(
      document as unknown as OrganizationDocument,
    );

    return org;
  }
```

**Comment by j-vitali** [comment]
Basically a correct retrieve must look like:

--- 
<img width="1080" alt="Screenshot 2024-05-23 alle 17 00 22" src="https://github.com/Automattic/mongoose/assets/161858414/b195f4b5-f1bd-48a4-9a4f-6aca0e5c4456">

**Comment by vkarpov15** [comment]
I'm confused, it looks like `throw new Error('should be populated');` means you intentionally throw an error if `populate()` doesn't return any results. So why is it surprising that your server returns an HTTP 500 error if `populate()` doesn't return any results?
